Verify User Agent String

megangardner opened the pull request:
#### 6fd973139cba771c96fe23bd15b08187330b99f8
<pre>
Verify User Agent String
<a href="https://bugs.webkit.org/show_bug.cgi?id=253213">https://bugs.webkit.org/show_bug.cgi?id=253213</a>
rdar://106116260

Reviewed by Tim Horton.

* Source/WebCore/loader/FrameLoader.cpp:
(verifyUserAgent):
(WebCore::FrameLoader::userAgent const):

Canonical link: <a href="https://commits.webkit.org/261353@main">https://commits.webkit.org/261353@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d43b48d65c700a2b6e525e50380f90309d7e2e80

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111350 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20489 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43889 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/87/builds/3007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120151 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115305 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21856 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11583 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/87/builds/3007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117115 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16232 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99410 "2 api tests failed or timed out") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104018 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98188 "Build was cancelled. Recent messages:worker ews159 ready; Configured build; Validated change; OS: Monterey (12.6), Xcode: 14.0; Cleaned up git repository; Cleaned and updated working directory; Updated branch information; Checked out required revision; Identifier: 261339@main; Skipping applying patch since patch_id isn't provided; Checked out pull request; Downloaded built product; Extracted built product; Killed old processes; run-api-tests (cancelled)") | [⏳ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-WK1-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44858 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12990 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32397 "Build is being retried. Recent messages:worker ews169 ready; Configured build; Validated change; OS: Big Sur (11.7.2), Xcode: 12.5.1; Cleaned up git repository; Cleaned and updated working directory; Updated branch information; Checked out required revision; Identifier: 261341@main; Skipping applying patch since patch_id isn't provided; Checked out pull request; Downloaded built product; Extracted built product; Crash collection has quiesced; Killed old processes (retry)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86677 "14 api tests failed or timed out") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13502 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9400 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18945 "Built successfully") | [💥 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51978 "An unexpected error occured. Recent messages:OS: Monterey (12.6.3), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7871 "'git push ...'") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15459 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -159,6 +159,15 @@
+#else
+void verifyUserAgent(const String& userAgent)
+{
+    UNUSED_PARAM(userAgent);
```
Just leave off the param name and then you don't need this line

hortont424 commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2919,10 +2928,15 @@ String FrameLoader::userAgent(const URL& url) const
     InspectorInstrumentation::applyUserAgentOverride(m_frame, userAgent);
 
+    verifyUserAgent(userAgent);
+    
```
Neaten this all up: compute the return value into a local, then call verify, then return it.

megangardner committed:
Verify User Agent String
https://bugs.webkit.org/show_bug.cgi?id=253213
rdar://106116260

Reviewed by Tim Horton.

* Source/WebCore/loader/FrameLoader.cpp:
(verifyUserAgent):
(WebCore::FrameLoader::userAgent const):

Canonical link: https://commits.webkit.org/261353@main
